### PR TITLE
9 use prisma orm

### LINKED ISSUE #9: use prisma orm
# Description
We need data persistence with the backend, so we have to implement a data store with an ORM

## acceptance criteria
- configurable from .env
- dev mode in-memory or file like sqlite
- uses migrations
- declares 1 schemas: 
````
Optins {
    email : string, unique
    source : string, nullable
}
````

### PR DESCRIPTION
**Pull request by badido39**
to review need to install Prisma (both cli and lib) , and run the folowing command :

1- yarn add prisma
2- npx prisma generate  
this to generate client inside nodemodules/@prisma otherwise the prismaCient wont created it self
3- npx prisma migrate deploy

and then test the route http://localhost:8000/subscription/ with body 
{
	"email":"12345@john.com"
}

and test the result 
the logic dosnt accept double subscription

### CODE REVIEW

File: hoggar/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  updatedAt DateTime @updatedAt
+
+  username String @unique
+  hash     String
+
+  firstName String?
+  lastName  String?
+
+  @@unique([id])
+}
```
Comment (osminosm) [inline hoggar/prisma/schema.prisma]: out of scope
```suggestion
```

File: hoggar/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+model Subscription {
+  id        String   @id @default(uuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  email String @unique
```
Comment (osminosm) [inline hoggar/prisma/schema.prisma:18]:
```suggestion
  email String @unique
  source String
```

File: hoggar/services/addSubscription.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    const subscriber = await prisma.subscription.create({
+      data: data,
+    });
+    return subscriber;
+  } catch (error) {
+    if (error instanceof PrismaClientKnownRequestError) {
+      if (error.code === "P2002") {
+        return "already subscribed !!!";
+      }
```
Comment (osminosm) [inline hoggar/services/addSubscription.ts]: should return error code not : message for human
```suggestion
      if (error.code === "P2002") {
        throw new Error("duplicate_email");
      }

```

File: package.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "build:icosium": "cd icosium && yarn build",
     "test:icosium": "cd icosium && yarn test",
     "eject:icosium": "cd icosium && yarn eject",
-    "dev:hoggar": "cd hoggar && yarn dev",
+    "dev:hoggar": "cd hoggar &&  yarn dev",
```
Comment (osminosm) [inline package.json]: - add prisma client as dependecy
- scripts to generate and migrate on root package.json
- add steps in DOC

**Review by osminosm — APPROVED** [review]

**Comment by nazimboudeffa** [comment]
I f you can explain a little bit more about those command, migrate and generate please
As in mongodb I didn't have found this kind of commands

**Review by nazimboudeffa — APPROVED** [review]

**Review by raxrs — APPROVED** [review]

**Pull request merged**